src/CoreShop/Bundle/CoreBundle/EventListener/NotificationRules/CustomerListener.php

```diff
@@ -39,14 +43,82 @@ public function applyRegisterCustomerRule(GenericEvent $event)
     {
         Assert::isInstanceOf($event->getSubject(), CustomerInterface::class);
 
+        /**
+         * @var $user CustomerInterface
+         */
         $user = $event->getSubject();
 
-        if($user->getIsGuest() === true) {
+        if ($user->getIsGuest() === true) {
             return;
         }
 
         $this->rulesProcessor->applyRules('user', $user, [
-            'type' => 'register',
+            'type' => UserTypeChecker::TYPE_REGISTER,
+            'recipient' => $user->getEmail(),
+            '_locale' => $this->shopperContext->getLocaleCode()
+        ]);
+    }
+
+    /**
+     * @param RequestNewsletterConfirmationEvent $event
+     */
+    public function applyNewsletterConfirmRequestRule(RequestNewsletterConfirmationEvent $event)
+    {
+        Assert::isInstanceOf($event->getCustomer(), CustomerInterface::class);
+
+        /**
+         * @var $user CustomerInterface
+         */
+        $user = $event->getCustomer();
+
+        if ($user->getIsGuest() === true) {
+            return;
+        }
+
+        if (!$user instanceof Concrete) {
+            return;
+        }
+
+        $newsletterTool = new Newsletter($user::classId());
```

Comment:
we should avoid using the pimcore newsletter since they're working with versions and stuff which could get messy if a customer (or a hook) triggers another update during the registration process

Reply:
done